src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -174,10 +174,6 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
 
             private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
             {
-                // First see if this is a document/project removal.  If so, clear out any state we
-                // have associated with any analyzers we have for that document/project.
-                ProcessRemovedDiagnostics(e);
```

Comment:
I think aggregate tagger is already per subject buffer.

IDiagnosticService is aggregating service. since error list needs to show errors from multiple workspace (VSWorkspace and MiscWorkspace), service can't be workspace service which is specific to only one workspace.

id of diagnostic is opaque. can't force it to be unique across workspace.